[#14] AWS S3 파일 삭제 기능 추가

cold-pumpkin opened the pull request:
Closes #14 
### AWS S3 파일 삭제 기능 추가

- [X] AWS S3 파일 삭제 기능 추가
- [X] AWS S3에 특정 파일이 존재하는 확인하기 위한 HEAD 요청 기능 추가
- [X] AWS S3의 파일 삭제 후 정상적으로 삭제되었는지 확인하는 테스트 코드 작성

cold-pumpkin committed:
Merge branch 'develop' of https://github.com/f-lab-edu/share-my-hobby into feature/#14

cold-pumpkin committed:
Merge branch 'feature/#12' of https://github.com/f-lab-edu/share-my-hobby into feature/#14

cold-pumpkin committed:
[#14] AWS S3 파일 삭제 기능 추가

cold-pumpkin committed:
[#14] AWS S3 삭제 테스트를 위한 HEAD 요청 기능 구현

cold-pumpkin committed:
[#14] AWS S3 파일 삭제 기능 테스트 코드

cold-pumpkin committed:
[#18] 유저 프로필 업데이트 관련 컨트롤러 및 서비스 추가

cold-pumpkin committed:
[#18] 유저 프로필 업데이트 매퍼 추가

cold-pumpkin committed:
[#18] 프로필 테이블 유니크 키 추가

- 유저 한명 당 하나의 프로필 data를 갖도록 user_id 외래키에 유니크
  키 추가

cold-pumpkin committed:
[#18] 프로필 상태메시지 업데이트 테스트 코드 작성

f-lab-dev commented on src/main/java/flab/project/sharemyhobby/aws/S3Uploader.java:
```diff
@@ -29,6 +31,18 @@ public String upload(MultipartFile uploadImage) throws IOException {
+    }
+
+    @Override
+    public void head(String originalFileName) {
```
S3가 아닌 로컬파일 구현체일 경우 이 head 메소드는 어떻게 구현될까요?

cold-pumpkin replied:
@f-lab-dev 아 생각해보니 이 `head(..)` 메소드는 S3에서 제공하는 기능이라 인터페이스에 명시해두는 것은 좋지 않을 것 같습니다..! 음 사실 `head(..)`를 구현했던 이유는 테스트코드 안에서 S3에 파일이 존재하는지 확인하기 위한 용도였는데요, 그럼 인터페이스에 명시된 `head(..)`는 지우고, 구현체인 `S3Uploader`에서는 `head(..)`를 `private`으로 변경하는게 좋을까요?

f-lab-dev replied:
네 내부에서만 사용하도록 하면 될 것 같아요~

cold-pumpkin replied:
넵 그렇게 수정해보겠습니다!

cold-pumpkin committed:
[#18] 리뷰반영: 삭제요청 시 파라미터로 사용될 이미지 파일명 칼럼 추가

cold-pumpkin committed:
[#18] 리뷰반영: 서비스에서 기존 프로필 찾아와 삭제하도록 로직 수정

cold-pumpkin committed:
[#18] 리뷰반영: 프로필 업데이트 테스트코드 수정 및 추가

cold-pumpkin committed:
[#18] 리뷰반영: 예외클래스 내용 오류 수정

cold-pumpkin committed:
[#18] 리뷰반영: 이미지파일명 DB에 저장하는 방식에서 URL에서 추출하는 방식으로 변경

cold-pumpkin committed:
Merge pull request #19 from f-lab-edu/feature/#18

[#18] 유저 프로필 수정 기능 추가

cold-pumpkin committed:
[#14] 리뷰반영: 파일 존재 여부 체크 로직 변경

- AWS S3 고유 메소드 head() 인터페이스에서 삭제
- 파일 존재여부 체크 메서드 인터페이스에 추가
- AWS S3 파일 존재여부 체크 메서드 구현

f-lab-dev commented on src/main/java/flab/project/sharemyhobby/util/FileUploader.java:
```diff
@@ -10,6 +10,6 @@
     void delete(String originalFileName);
 
-    void head(String originalFileName);
+    boolean checkExist(String originalFileName);
```
boolean을 리턴하는 메소드는 보통 `is`를 앞에 붙입니다~ `isExist`가 나을 것 같네요

cold-pumpkin replied:
아하 네 수정하겠습니다~

cold-pumpkin committed:
[#14] 리뷰반영: boolean 타입 리턴 메소드 이름 수정

cold-pumpkin committed:
[#14] merge conflict 해결
